plugins/modules/site_workflow_manager.py

```diff
@@ -929,48 +1652,123 @@ def get_diff_deleted(self, config):
             This method initiates the deletion of a site by calling the 'delete_site' function in the 'sites' family
             of the Cisco Catalyst Center API. It uses the site ID obtained from the 'have' attribute.
         """
-
         site_exists = self.have.get("site_exists")
         site_name = self.want.get("site_name")
+
         if not site_exists:
             self.status = "success"
             self.site_absent_list.append(site_name)
-            self.log("Unable to delete site '{0}' as it's not found in Cisco Catalyst Center".format(site_name), "INFO")
+            self.log(
+                "Unable to delete site '{0}' as it's not found in Cisco Catalyst Center".format(site_name), "INFO")
             return self
 
-        # Check here if the site have the childs then fetch it using get membership API and then sort it
-        # in reverse order and start deleting from bottom to top
-        site_id = self.have.get("site_id")
-        mem_response = self.dnac._exec(
-            family="sites",
-            function="get_membership",
-            op_modifies=True,
-            params={"site_id": site_id},
-        )
-        self.log("Received API response from 'get_membership': {0}".format(str(mem_response)), "DEBUG")
-        site_response = mem_response.get("site").get("response")
-        self.log("Site {0} response along with it's child sites: {1}".format(site_name, str(site_response)), "DEBUG")
+        if self.get_ccc_version_as_integer() <= self.version_2_3_5_3:
+            site_id = self.have.get("site_id")
+            mem_response = self.dnac._exec(
+                family="sites",
+                function="get_membership",
```

Comment:
we already have this API in dnac.py. Please check

Reply:
In dnac.py, the Membership API was originally used only to return the deviceId. I have now modified this function to retrieve the full response (which includes checking the site hierarchy) and the deviceId as well